### PR TITLE
[MDS-4508] NOD contacts Database/Api updates

### PR DESCRIPTION
**Pull request by matbusby-fw**
## Objective 

[MDS-4508](https://bcmines.atlassian.net/browse/MDS-4508)

- Created a migration to add a notice of departure contact table
- added notice of departure contact model to the api
- updated create nod endpoint to include the new contact

_Why are you making this change? Provide a short explanation and/or screenshots_

### CODE REVIEW

File: migrations/sql/V2022.06.23.09.30__create_notice_of_departure_contacts_table.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
+CREATE TABLE IF NOT EXISTS notice_of_departure_contact
+(
+    notice_of_departure_contact_guid uuid DEFAULT gen_random_uuid() PRIMARY KEY,
```
Comment (hitankar-freshworks) [inline migrations/sql/V2022.06.23.09.30__create_notice_of_departure_contacts_table.sql]:
```suggestion
    nod_contact_guid uuid DEFAULT gen_random_uuid() PRIMARY KEY,
```
the col names are going out of hand. suggest smaller

File: migrations/sql/V2022.06.23.09.30__create_notice_of_departure_contacts_table.sql [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    notice_of_departure_guid         uuid NOT NULL,
```
Comment (hitankar-freshworks) [inline migrations/sql/V2022.06.23.09.30__create_notice_of_departure_contacts_table.sql]:
```suggestion
    nod_guid         uuid NOT NULL,
```
for the sake of consistency

File: services/core-api/app/api/notice_of_departure/models/notice_of_departure.py [hunk 3]
```diff
@@ -74,6 +75,10 @@ class NoticeOfDeparture(SoftDeleteMixin, AuditMixin, Base):
         primaryjoin=
         "and_(NoticeOfDeparture.nod_guid==NoticeOfDepartureDocumentXref.nod_guid, NoticeOfDepartureDocumentXref.deleted_ind==False)",
         order_by='desc(NoticeOfDepartureDocumentXref.create_timestamp)')
+    contacts = db.relationship(
+        'NoticeOfDepartureContact',
+        lazy='select',
+        primaryjoin="and_(NoticeOfDeparture.nod_guid==NoticeOfDepartureContact.notice_of_departure_guid, NoticeOfDepartureContact.deleted_ind==False)", )
```
Comment (hitankar-freshworks) [inline services/core-api/app/api/notice_of_departure/models/notice_of_departure.py]: nice, we are lazyloading. I think we will need to indicate joining contacts when `find_one` method is called

File: services/core-api/app/api/notice_of_departure/resources/notice_of_departure_list.py [hunk 4]
```diff
@@ -113,6 +111,35 @@ def post(self):
             location='json',
             choices=list(NodStatus),
             store_missing=False)
+        parser.add_argument(
```
Comment (hitankar-freshworks) [inline services/core-api/app/api/notice_of_departure/resources/notice_of_departure_list.py:114]: I had a different approach in mind

File: services/core-api/app/api/notice_of_departure/models/notice_of_departure.py (hunk 3 above)
Comment (matbusby-fw) [reply services/core-api/app/api/notice_of_departure/models/notice_of_departure.py]: Yeah, this is the part I'm looking at now.  We can chat about it.

File: services/core-api/app/api/notice_of_departure/resources/notice_of_departure_list.py (hunk 4 above)
Comment (matbusby-fw) [reply services/core-api/app/api/notice_of_departure/resources/notice_of_departure_list.py:114]: I'm all 👂 👂

**Pull request closed without merging**